lib/install.js

```diff
@@ -216,9 +217,46 @@ function Installer (where, dryrun, args) {
 }
 Installer.prototype = {}
 
-Installer.prototype.run = function (cb) {
+function updateStatsFile (itWorked) {
+  // If the stats reporter hasn't managed to PUT yet then kill it so that it doesn't
+  // step on our updating the cli-stats json
+  if (npm.statsProcess) npm.statsProcess.kill('SIGKILL')
+  var statsFile = path.join(npm.config.get('cache'), 'anonymous-cli-stats.json')
+  var stats
+  try {
+    stats = JSON.parse(fs.readFileSync(statsFile))
```

Comment:
I think that the latest refactor is more than sufficient for now. Thanks for the clear answers!